src/Raven.Server/Rachis/ClusterTransactionWaiter.cs

```diff
@@ -2,56 +2,60 @@
 using System.Collections.Concurrent;
 using System.Threading;
 using System.Threading.Tasks;
+using Sparrow.Json.Parsing;
 
-namespace Raven.Server.Documents
+namespace Raven.Server.Rachis
 {
-    public class ClusterTransactionWaiter
+    public class ClusterTransactionWaiter : AsyncWaiter<ClusterTransactionCompletionResult>
```

Comment:
What is the question?
If we need the inheritance?
The question is if we want to separate the result from the `DocumentDatabase` to the result from the `ClusterStateMachine`.
For now, I used the same collection for both of them so actually, I can remove the inheritance and define directly the tasks to return  `ClusterTransactionCompletionResult` (anyway I think it is prettier like that but it is not important if you don't)

Anyway, as it is implemented now I have to use `TrySet` at least in the `DocumentDatabase` because there can be two attempts to set the result. 
Or I can conditionally set it only if it is not a shared database. does it sound good?